lib/needle.js

```diff
@@ -914,3 +932,6 @@ module.exports.defaults = function(obj) {
 module.exports.request = function(method, uri, data, opts, callback) {
   return new Needle(method, uri, data, opts, callback).start();
 };
+
+// Exporting for testing
+module.exports.should_proxy_to = should_proxy_to;
```

Comment:
This seems a bit dirty.

Reply:
@tomas Thank you for your review!
The only other option I see is introducing a new file `should_proxy_to.js` that exposes `should_proxy_to` and will be used in `needle.js` and in the tests.
Both options are not very clean, I thought that this one was cleaner.

I've also rewritten the description for the PR and got rid of the dependency, introducing a regex instead.
What do you think?

Reply:
Thanks for removing the dependency.

As for the tests, what's wrong with the current method we're using to test the NO_PROXY env var? (Setting ` process.env.NO_PROXY` and then removing it once the test passes)

Reply:
Hi @tomas, any thoughts on this PR? I'd happily change anything else you see fit to get it merged and solve this issue